src/aliased_buffer.h

```diff
@@ -129,7 +129,7 @@ class AliasedBuffer {
 
     template <typename T>
     inline Reference& operator=(const T& val) {
-      aliased_buffer_->SetValue(index_, val);
+      aliased_buffer_->SetValue(index_, static_cast<NativeT>(val));
```

Comment:
I agree that this does the same as the original code, but I'd prefer a solution where these operators aren't templates and the caller has to think about whether casting is valid. Otherwise we hide all of the danger inside this function instead of getting useful compiler warnings elsewhere.

Reply:
you'd still get compiler warnings w/ static_cast right?  I don't remember why these functions were templated instead of taking a `NativeT` as a param.

Reply:
No, the explicit cast tells the compiler that you've carefully thought about what happens here and are okay with a possibly lossy conversion.

---
In reply to: [177183603](https://github.com/nodejs/node-chakracore/pull/503#discussion_r177183603) [](ancestors = 177183603)

Reply:
I see.  So, it would be interesting to see if we could remove the templated functions here & push casts onto the caller.  I don't recall why I did it this way previously.